mediaflow_proxy/utils/packed.py

```
@@ -142,18 +142,19 @@ class UnpackingError(Exception):
-                match = re.search(pattern, unpacked_code)
-                if match:
-                    m3u8_url = match.group(1)
-                    return m3u8_url
+                for pattern in patterns:
+                    match = re.search(pattern, unpacked_code)
+                    if match:
+                        m3u8_url = match.group(1)
+                        return m3u8_url
     except Exception as e:
         logger.error("Eval solver error\n",e)
         raise Exception("Error in eval_solver")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)